### PR TITLE
feat: created the OG image for repository pages

### LINKED ISSUE #3098: Feature: Open Graph (OG) image for repository pages
As a user of repository pages, I want an open graph (OG) image for a repository page I visit so that it is shareable with more context on social media platforms, Slack, etc.

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Implements the open graph (OG) image for repository pages.

## Related Tickets & Documents

Closes #3098

## Mobile & Desktop Screenshots/Recordings

image

## Steps to QA

1. Go to e.g. https://deploy-preview-3117--oss-insights.netlify.app/s/kubernetes/kubernetes
2. The repository page loads.
3. In Slack or some social network, paste that URL.
4. When the post or message is published, an OG image appears.

CleanShot 2024-04-04 at 11 50 15

## Tier (staff will fill in)

- [ ] Tier 1
- [x] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

File: components/Repositories/RepositoryOgImage.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ogImageUrl: string;
+}
+
+export const RepositoryOgImage = ({ repository, ogImageUrl }: WorkspaceOgImageProps) => {
+  return (
+    <SEO
+      title={`Repositories | ${repository.full_name}`}
+      description={repository.description ?? ""}
+      image={ogImageUrl}
+      twitterCard="summary_large_image"
```
Comment (nickytonline) [inline components/Repositories/RepositoryOgImage.tsx:24]: Even though this is set, the rendered meta tag is showing "summary", the small version. Looking into this.

Comment (nickytonline) [reply components/Repositories/RepositoryOgImage.tsx:24]: Fixed in 1e3da04a1dd1f3eb25ebaf0f803a69efd1da2101.

**Comment by bdougie** [comment]
Looking good on imessage. Title does get cut off though.

Screen Shot 2024-04-04 at 9 59 46 AM

**Comment by nickytonline** [comment]
> Looking good on imessage. Title does get cut off though.
> 
> Screen Shot 2024-04-04 at 9 59 46 AM

@isabensusan should we reduced the font size or is the avatar and just the name not org better?

**Review by brandonroberts — APPROVED** [review]

**Review by jpmcb — APPROVED** [review]

**Comment by brandonroberts** [comment]
For reference, GitHub wraps the name

image

**Comment by nickytonline** [comment]
Yeah, chatted with @isabensusan, and that's what I'm going to do.

**Comment by nickytonline** [comment]
So with a short and long name.

image

image

File: netlify/edge-functions/repo-page-card.tsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+import { ImageResponse } from "og_edge";
+import type { Config } from "https://edge.netlify.com";
+
+const baseApiUrl = Deno.env.get("NEXT_PUBLIC_API_URL");
+
+type StatsType<T = "stars" | "forks"> = T extends "stars"
+  ? { star_count: number; bucket: string }
+  : { forks_count: number; bucket: string };
+
+function humanizeNumber(num: number | string, type: "comma" | "abbreviation" | null = null) {
```
Comment (nickytonline) [inline netlify/edge-functions/repo-page-card.tsx:11]: Post release, I will refactor the use of the shared functions that are currently copy pasta.

**Pull request merged**